mixer/adapter/stackdriver/metric/metric_test.go

```diff
@@ -122,6 +122,38 @@ func TestFactory_NewMetricsAspect_Errs(t *testing.T) {
 	}
 }
 
+func TestMetricType(t *testing.T) {
+	cfg := &config.Params{
+		MetricInfo: map[string]*config.Params_MetricInfo{
+			"metric1": {
+				MetricType: "istio.io/metric1",
+			},
+			"metric2": {},
+		},
+	}
+	metrics := map[string]*metrict.Type{
+		"metric1": {},
+		"metric2": {},
+	}
+	b := &builder{createClient: clientFunc(nil)}
+	b.SetMetricTypes(metrics)
+	b.SetAdapterConfig(cfg)
+	env := test.NewEnv(t)
+	h, err := b.Build(context.Background(), env)
+	if err != nil {
+		t.Fatalf("Failed building metric handler: %v", err)
+	}
+
+	expected := map[string]bool{"istio.io/metric1": true, customMetricPrefix + "metric2": true}
```

Comment:
instead of making an expected map and got map, let's just make two test cases:

```
case := []struct{
    name string
    cfg *config.Params
    want string
}{
    {"metric type", paramsWithMetricInfo, "istio.io/metric1"},
    {"no metric type", &config.Params{}, customMetricPrefix + "metric2"}
}

for _, tc := range cases {
   t.Run(tc.name, func(t *testing.T) {
      b.SetAdapterConfig(tc.cfg)
      h, _ := b.Build(...)
      ...
      if got != tc.want {
          t.Errorf("Bad metric type. Got %s, wanted %s", got, tc.want)
      }
   })
}
```